exportlib.php

```diff
@@ -180,34 +182,154 @@ private function quote($str)
+                , PARAM_FILE));
+
+        $writer->add_data([
+            'Feedbackid: ' . $this->feedback->id,
+            'Feedbackid: ' . $this->feedback->name
+        ]);
+
+        // Get questions
+        $qus = block_coursefeedback_get_questions_by_language($this->feedback->id, [current_language()]);
+        $questions = null;
+
+        if ($this->questionid) {
+            // Only display one question.
+            foreach ($qus as $qu) {
+                if ($qu->questionid == $this->questionid) {
+                    $questions = array($qu);
+                }
+            }
+        } else {
+            // Display all questions.
+            $questions = $qus;
+        }
+
+        foreach ($questions as $question) {
+            // Output headings.
+            $writer->add_data([]);
+            $writer->add_data([
+                $question->question,
+                $question->questionid
+            ]);
+
+            $writer->add_data([
+                get_string('course'),
+                get_string('user'),
+                get_string('name'),
+                get_string('categories'),
+                get_string('categorypath', 'block_coursefeedback'),
+                get_string('notif_emoji_super', 'block_coursefeedback'),
+                get_string('notif_emoji_good', 'block_coursefeedback'),
+                get_string('notif_emoji_ok', 'block_coursefeedback'),
+                get_string('notif_emoji_neutral', 'block_coursefeedback'),
+                get_string('notif_emoji_bad', 'block_coursefeedback'),
+                get_string('notif_emoji_superbad', 'block_coursefeedback'),
+                get_string('table_html_average', 'block_coursefeedback'),
+                get_string('table_html_votes', 'block_coursefeedback'),
+                get_string('table_html_nochoice', 'block_coursefeedback'),
+            ]);
+
+            // Get courseids and the amount of answers in this course for the current question.
+            $params = array(
+                'questionid' => $question->questionid,
+                'feedbackid' => $this->feedback->id,
+                'answerlimit' => 0,
+                'feedbackid2' => $this->feedback->id,
+                'questionid2' => $question->questionid,
+            );
+            $sql = "
+                SELECT one.courseid, five.usero, c.shortname, c.category, cc.path,  
+                    six.one, six.two, six.three, six.four, six.five, six.six, 
+                    (six.answersum / ( NULLIF ((one.anstotal - six.abstain), 0))) as average, 
+                    one.anstotal, six.abstain FROM
+                    (
+                        SELECT course as courseid, count(*) as anstotal FROM {block_coursefeedback_answers}
+                            WHERE questionid = :questionid AND coursefeedbackid = :feedbackid
+                            GROUP BY course
+                            HAVING count(*) > :answerlimit
+                    ) one
```

Comment:
Bitte verwende andere Aliase für die Tabellen. Du verwendest ja bereits Zahlwörter für die Spaltennamen weiter unten.
```suggestion
                    ) courseanswers
```

Reply:
Auch auf der Liste